internal/precache/workload/pullImages.go

```diff
@@ -188,3 +188,19 @@ func ValidatePrecache(status *precache.Progress, bestEffort bool) error {
 	}
 	return nil
 }
+
+func Precache(precacheSpec []string, authFile string, bestEffort bool) error {
+	// Pre-cache images
+	status, err := PullImages(precacheSpec, authFile)
+	if err != nil {
```

Comment:
Seems that you return an error here also when `bestEffort` is true...

Reply:
This is the logic that IBU have, we can change it for IBI

Reply:
OK, seems that PullImages never return an error, it's always nil.